### ISSUE TITLE
fix(model): reject MTP prefix restore across target execution identities

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12865]
# fix(model): reject MTP prefix restore across target execution identities

```routing
lane: model
paths: ["internal/model/kv.go","internal/model/qwen35_mtp_snapshot_identity_test.go"]
expected_steps: 3
```

## Parent context
Parent: #12238. Coordinates with: #12331, #12859, #12856.
Promotion requires: #12239.

## Current state
PrefixSnapshot carries exact raw-hidden/token lineage and clones it, but Restore checks backend identity only. It does not reject another model or precision mode before transferring cached history. #12856 scopes structural sharing; this is a separate execution identity correctness boundary with no COW changes.

## Why this is next
Complete captured history is a prerequisite for safe native quantized MTP admission. History from a different model or numerical execution mode must not become valid merely because token IDs and dimensions match.

## Working spine
Actual F32 hybrid prompt capture -> independently owned PrefixSnapshot clones -> warm target restore -> divergent suffix Step -> exact lineage/raw hidden/continuation; mismatched model or precision restore rejects atomically.

## Core through-line
Bind snapshot restore to the source model/execution identity while preserving correct same-mode clones and existing target-history validation. Do not widen quantized or Metal admission.

## Scope
Exactly internal/model/kv.go and new internal/model/qwen35_mtp_snapshot_identity_test.go. Initial source-only reproduction edits only the new test. Three steps: actual snapshot regression, minimal identity check after behavioral RED, same focused GREEN/review/guarded landing. No12859 overlay required for F32 history witness.

## Gold-plating boundary
No COW, quant/Metal admission, graph, kernels, HTTP, global retention or performance claims. If identity completeness needs other files, rescope before edits. Keep existing warm-history and depth guards.

## Done condition
Two same-model/mode clones restore and independently continue with exact history and logits; source snapshot remains unchanged. Different-model/precision restore errors before destination or snapshot ownership mutation. All owners close. Behavioral RED then SAME GREEN and guarded landing.

## Definition of done
- [ ] Reviewed actual-state failing regression and deterministic command.
- [ ] Behavioral RED/SAME GREEN and independently witnessed exact-path landing.

## Witness
GOMAXPROCS=2 go test -p=1 ./internal/model -run '^TestQwen35MTPSnapshotExecutionIdentity$' -count=1 -timeout=60s

## Acceptance gate
Actual constructor, Prefill, PrefixSnapshot.Clone/Restore, Step, raw-history and token-lineage assertions. No fabricated capture. Rejected restores leave both source and destination unchanged. NoCPU until campaign grant, compilecap300s.

## Lane
model

## Closure binding
Close only bounded child through verified exact-path (fak model) commit.12238 native admission and original physical acceptance remain open.

## Work estimate
Estimate: 1 point, provisional bounded correctness leaf.

## Overall completion contribution
Contribution: 1/12 points inherited from12238 provisional planning baseline; not authoritative completion credit.

## Completion standard
development

## Problem frame
Centrality: Core
P1: advanced - bind raw hidden lineage to execution identity.
P2: preserved - no speed claim.
P3: preserved - twofile scope and existing deep clone.
P4: advanced - actual restore/continuation proof.
